samples/bluetooth/broadcast_audio_source/src/main.c

```diff
@@ -151,25 +225,33 @@ static void stream_sent_cb(struct bt_bap_stream *stream)
 
 	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
 #if defined(CONFIG_LIBLC3)
-	int lc3_ret;
 	uint8_t lc3_encoder_buffer[preset_16_2_1.qos.sdu];
 
 	if (lc3_encoder == NULL) {
 		printk("LC3 encoder not setup, cannot encode data.\n");
 		return;
 	}
 
-	lc3_ret = lc3_encode(lc3_encoder, LC3_PCM_FORMAT_S16, mock_data, 1, octets_per_frame,
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+	int size = ring_buf_get(&audio_ring_buf, (uint8_t *)send_pcm_data, sizeof(send_pcm_data));
+
+	if (size < sizeof(send_pcm_data)) {
+		printk("Not enough bytes ready, padding %d!\n", sizeof(send_pcm_data) - size);
+		memset(&((uint8_t *)send_pcm_data)[size], 0, sizeof(send_pcm_data) - size);
+	}
+#endif
+
+	ret = lc3_encode(lc3_encoder, LC3_PCM_FORMAT_S16, send_pcm_data, 1, octets_per_frame,
```

Comment:
Calling such heavy function inside bluetooth callback that is executed with cooperative thread priority (meaning other workqueues or threads cannot pre-empt the execution even if they have higher priority) is probably the reason why you observe data sizes other than 192 bytes in `data_received()`.

I believe doing the encoding in preemptible thread will solve the USB issues.

Reply:
Thanks a lot for your finding and hint on solution! - working on offloading the encoding to a separate thread (moved the PR to draft for now)